modules/hal_rpi_pico/CMakeLists.txt

```diff
@@ -130,14 +145,15 @@ if(CONFIG_HAS_RPI_PICO)
     COMPILE_FLAGS $<TARGET_PROPERTY:compiler,warning_no_pointer_arithmetic>
   )
 
-  zephyr_library_sources_ifdef(CONFIG_PICOSDK_USE_PIO
-          ${rp2_common_dir}/hardware_pio/pio.c)
-  zephyr_include_directories_ifdef(CONFIG_PICOSDK_USE_PIO
-          ${rp2_common_dir}/hardware_pio/include)
-
   zephyr_library_sources_ifdef(CONFIG_PICOSDK_USE_CLAIM
-          ${rp2_common_dir}/hardware_claim/claim.c)
+          ${common_dir}/hardware_claim/claim.c)
   zephyr_include_directories_ifdef(CONFIG_PICOSDK_USE_CLAIM
-          ${rp2_common_dir}/hardware_claim/include)
+          ${common_dir}/hardware_claim/include)
+
+  zephyr_library_sources_ifdef(CONFIG_SOC_RP2350
```

Comment:
This file is not a driver, but OS/Middleware, so I would rather not import it if possible.
It would probably be better to reimplement it in soc.c on the Zephyr side.